### PR TITLE
[Dashboard] Feature: Add initial insight UI

### PR DESCRIPTION
**Pull request by AmineAfia**
## Problem solved
BLOCK-436

This PR adds an initial UI for insight to give visibility to insight in the platform. The first three blueprints are Data APIs available for all users. The next iteration will add the ability to create a new blueprint.

---

## PR-Codex overview
This PR introduces a new `Insight` page under the project structure, adding components for displaying blueprints, including a header and an explorer. It also implements authentication redirection and a loading state for better user experience.

### Detailed summary
- Added a new route for `Insight` in `layout.tsx`.
- Created `BlueprintsPageHeader` component for the `Insight` page.
- Implemented `Page` component with authentication check and redirection logic.
- Developed `BlueprintsPage` component to display blueprints with a loading state.
- Created `getProjectBlueprints` function to fetch blueprint data.
- Added `BlueprintsExplorer` component to list blueprints with a fallback message.
- Implemented `BlueprintCard` for individual blueprint display.
- Included loading spinner in `Loading` component.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

Linear Ticket: https://linear.app/thirdweb/issue/BLOCK-436/implement-initial-insight-ui

---

## PR-Codex overview
This PR introduces a new `Insight` page within the project dashboard, allowing users to view and manage blueprints. It includes authentication checks, a header component, and displays a list of available blueprints with relevant information.

### Detailed summary
- Added route for `Insight` page.
- Implemented authentication check in `Page` component.
- Created `BlueprintsPageHeader` for the `Insight` page.
- Developed `BlueprintsPage` to render header and blueprints.
- Added `getProjectBlueprints` function to fetch blueprint data.
- Implemented `BlueprintsExplorer` to display blueprints.
- Created `BlueprintCard` for individual blueprint representation.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

Linear Ticket: https://linear.app/thirdweb/issue/BLOCK-436/implement-initial-insight-ui

---

## PR-Codex overview
This PR introduces a new `Insight` page in the application, featuring blueprints for transaction data, events, and tokens. It includes authentication checks, a header component, and a display for available blueprints.

### Detailed summary
- Added `Insight` route to the application.
- Created `Page` component with authentication redirection.
- Introduced `BlueprintsPageHeader` for the `Insight` page.
- Implemented `BlueprintsPage` to display `BillingAlerts` and blueprints.
- Added `BlueprintsExplorer` to list blueprints with a card interface.
- Defined `getProjectBlueprints` function to retrieve mock blueprint data.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by AmineAfia** [comment]
* **#5435** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5435?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5435?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  return (
+    <div className="border-b pb-4">
+      <div className="flex flex-col gap-4 md:pb-4 lg:flex-row lg:justify-between">
+        <div>
+          <h1 className="mb-1.5 font-semibold text-3xl tracking-tight lg:text-4xl">
+            Blueprints
+          </h1>
+        </div>
+        <Button className="cursor-not-allowed gap-2 opacity-50" disabled>
+          <div className="flex items-center">
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx:16]: Remove this `div` to fix the spacing between icon and text

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    {
+      id: "3",
+      name: "Tokens",
+      slug: "tokens-blueprint",
+      description: "Query ERC-20, ERC-721, and ERC-1155 tokens",
+    },
+  ] as Blueprint[];
+}
+
+async function BlueprintsPageAsync() {
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx:46]: This component does not need to be async

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+export function BlueprintsPage(props: {
+  className?: string;
+}) {
+  return (
+    <div className={props.className}>
+      <BillingAlerts className="mb-10" />
+      <BlueprintsPageHeader />
+      <div className="h-6" />
+      <Suspense fallback={<Loading />}>
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx:16]: No suspense required

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+      description: "Query ERC-20, ERC-721, and ERC-1155 tokens",
+    },
+  ] as Blueprint[];
+}
+
+async function BlueprintsPageAsync() {
+  const blueprints = getProjectBlueprints();
+
+  return (
+    <ClientOnly ssr={<Loading />}>
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx:50]: Why is `ClientOnly` used here?

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsExplorer.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+function BlueprintCard(props: {
+  blueprint: Blueprint;
+}) {
+  const { blueprint } = props;
+  return (
+    <div
+      key={blueprint.id}
+      className="relative flex items-center gap-4 rounded-lg border border-border bg-muted/50 p-4 transition-colors hover:bg-muted/70"
+    >
+      <ProjectAvatar className="size-10 rounded-full" src="" />
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsExplorer.tsx:53]: Don't use `ProjectAvatar` here - this is not a project link. use a different icon

**Review by MananTank — CHANGES_REQUESTED** [review]
Please address comments

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsExplorer.tsx (hunk 5 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsExplorer.tsx]: done

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx (hunk 4 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx]: removed

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx (hunk 2 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPage.tsx]: removed async

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx (hunk 1 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx]: done

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+            <PlusIcon className="size-4" />
+            Create Blueprint (Coming Soon)
+          </div>
+        </Button>
+      </div>
+    </div>
+  );
```
[the remark this answers is not in the document]
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx:21]: done

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+
+export function BlueprintsPageHeader() {
+  return (
+    <div className="flex grow flex-col">
+      <div className="border-border border-b py-10">
+        <div className="container flex items-center justify-between">
+          <h1 className="font-semibold text-3xl tracking-tight">Blueprints</h1>
+          <Button className="cursor-not-allowed gap-2 opacity-50" disabled>
+            <PlusIcon className="size-4" />
+            Create Blueprint (Coming Soon)
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx:14]: This creates an Overflow on mobile

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+
+import { Button } from "@/components/ui/button";
+import { PlusIcon } from "lucide-react";
+
+export function BlueprintsPageHeader() {
+  return (
+    <div className="flex grow flex-col">
+      <div className="border-border border-b py-10">
+        <div className="container flex items-center justify-between">
+          <h1 className="font-semibold text-3xl tracking-tight">Blueprints</h1>
```
Comment (MananTank) [inline apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx:11]: The page's title "Blueprints" is confusing - "Insight" would be better

**Review by MananTank — CHANGES_REQUESTED** [review]
please address comments

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx (hunk 8 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx]: changed it

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx (hunk 7 above)
Comment (AmineAfia) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/insight/components/BlueprintsPageHeader.tsx:19]: fixed

**Review by MananTank — APPROVED** [review]

**Pull request merged**